src/test/java/de/bonndan/nivio/input/GroupQueryResolverTest.java

```diff
@@ -0,0 +1,97 @@
+
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class GroupQueryResolverTest {
+
+    private GroupQueryResolver groupResolver;
+
+    @Mock
+    private ProcessLog processLog;
+
+    @BeforeEach
+    public void setup() {
+        groupResolver = new GroupQueryResolver(processLog);
+    }
+
+    @Test
+    void process_doesNotContainDuplicatedItemsInGroups() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+        item.setGroup("groupIdentifier");
+        landscape.setItems(Set.of(item));
+        landscape.addGroup(new Group("groupIdentifier"));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+        groupDescription.setIdentifier("groupIdentifier");
+        groupDescription.setContains(List.of("itemidentifier"));
+        input.getGroups().put("group", groupDescription);
+
+        groupResolver.process(input, landscape);
+
+        assertThat(landscape.getItems().stream().count()).isEqualTo(1L);
+        assertThat(landscape.getGroup("groupIdentifier").getItems()).containsExactly(item);
+    }
+
+    @Test
+    void process_findsItemsIfQueryConditionIsNotLowercase() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+        item.setGroup("groupIdentifier");
+        landscape.setItems(Set.of(item));
+        landscape.addGroup(new Group("groupIdentifier"));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+        groupDescription.setIdentifier("groupIdentifier");
+        groupDescription.setContains(List.of("itemIdentifier"));
+        input.getGroups().put("group", groupDescription);
+
+        groupResolver.process(input, landscape);
+
+        assertThat(landscape.getItems().stream().count()).isEqualTo(1L);
+        assertThat(landscape.getGroup("groupIdentifier").getItems()).containsExactly(item);
+    }
+
+    @Test
+    void process_containsCommonGroupIfGroupWasEmpty() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+
+        landscape.setItems(Set.of(item));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+
+        input.getGroups().put("group", groupDescription);
+
+        groupResolver.process(input, landscape);
+
+        assertThat(landscape.getItems().stream().count()).isEqualTo(1L);
+        assertThat(landscape.getGroups().values().size()).isEqualTo(1);
```

Comment:
This aims for the las stream of the `process()` method. The groups are not updated on the landscape, only `getGroup(somethingInvalid)` returns `Group.COMMON`. Is this intentional?

Reply:
Has been refactored.